### PR TITLE
fix: JSON deserialization not firing variable create events for blocks

### PR DESCRIPTION
**Pull request by bsampada7**
## The basics

- [x ] I [validated my changes](https://developers.google.com/blockly/guides/contribute/core#making_and_verifying_a_change)

### Resolves

Fixes https://github.com/google/blockly/issues/7599

### Proposed Changes

Add code to fire VAR_CREATE event when a JSON is serialized and new variables are created while creating a block.

### Reason for Changes

Fixes the bug of variable create event not being fired, even though a variable is created when a block is created through JSON serialization.

### Test Coverage

Manually tested that the VAR_CREATE event fires when a variable is created during JSON serialization.

### Documentation

N/A

### Additional Information

N/A

### CODE REVIEW

File: core/serialization/blocks.ts [hunk 1]
```diff
@@ -425,6 +427,19 @@ export function appendInternal(
...
+    );
+    // Fire a VarCreate event for each (if any) new variable created.
+    for (let i = 0; i < newVariables.length; i++) {
+      const thisVariable = newVariables[i];
+      eventUtils.fire(
+        new (eventUtils.get(eventUtils.VAR_CREATE))(thisVariable),
+      );
+    }
+    // Block events come after var events, in case they refer to newly created
+    // variables.
```
Comment (BeksOmega) [inline core/serialization/blocks.ts:442]: Would you mind separating this into a separate function?

**Review by BeksOmega — CHANGES_REQUESTED** [review]
This looks great thank you for the fix! Just one organizational change then should be good to go =)

File: core/serialization/blocks.ts (hunk 1 above)
Comment (bsampada7) [reply core/serialization/blocks.ts]: Sure, just extracted it to a separate function

**Review by BeksOmega — APPROVED** [review]
Wonderful! Thank you so much for the fix :D

**Pull request merged**